src/components/customers/CustomerPaymentsList.tsx

```diff
@@ -0,0 +1,142 @@
+import { payablePaymentStatusMapping } from '~/core/constants/statusInvoiceMapping'
+import { intlFormatNumber } from '~/core/formats/intlFormatNumber'
+import { CUSTOMER_PAYMENT_DETAILS_ROUTE } from '~/core/router'
+import { deserializeAmount } from '~/core/serializers/serializeAmount'
+import { formatDateToTZ } from '~/core/timezone'
+import { copyToClipboard } from '~/core/utils/copyToClipboard'
+import {
+  CurrencyEnum,
+  GetPaymentListQuery,
+  GetPaymentListQueryHookResult,
+  PaymentForPaymentsListFragment,
+  PaymentTypeEnum,
+} from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+
+interface CustomerPaymentsListProps {
+  payments: PaymentForPaymentsListFragment[]
+  loading: boolean
+  metadata?: GetPaymentListQuery['payments']['metadata']
+  fetchMore?: GetPaymentListQueryHookResult['fetchMore']
+}
+
+export const CustomerPaymentsList: FC<CustomerPaymentsListProps> = ({
+  payments,
+  loading,
+  metadata,
+  fetchMore,
+}) => {
+  const { translate } = useInternationalization()
+
+  return (
+    <InfiniteScroll
+      onBottom={() => {
+        const { currentPage = 0, totalPages = 0 } = metadata || {}
+
+        currentPage < totalPages &&
+          !loading &&
+          fetchMore?.({
+            variables: { page: currentPage + 1 },
+          })
+      }}
+    >
+      <Table
+        name="customer-payments-list"
+        data={payments}
+        containerSize={{ default: 4 }}
+        isLoading={loading}
+        actionColumn={() => {
+          return [
+            {
+              startIcon: 'duplicate',
+              title: translate('text_1737029625089rtcf3ah5khq'),
+              onAction: ({ id }) => {
+                copyToClipboard(id)
+                addToast({
+                  severity: 'info',
+                  translateKey: translate('text_17370296250897n2pakp5v33'),
+                })
+              },
+            },
+          ]
+        }}
+        actionColumnTooltip={() => translate('text_637f813d31381b1ed90ab326')}
+        onRowActionLink={(request) =>
+          generatePath(CUSTOMER_PAYMENT_DETAILS_ROUTE, {
+            paymentId: request.id,
+            customerId: request.customer.id,
+          })
+        }
+        columns={[
+          {
+            key: 'payablePaymentStatus',
+            title: translate('text_63ac86d797f728a87b2f9fa7'),
+            minWidth: 80,
+            content: ({ payablePaymentStatus }) => (
+              <Status
+                {...payablePaymentStatusMapping({
+                  payablePaymentStatus: payablePaymentStatus ?? undefined,
+                })}
+              />
+            ),
+          },
+          {
+            key: 'payable.__typename',
+            title: translate('text_63ac86d797f728a87b2f9fad'),
+            minWidth: 160,
+            maxSpace: true,
+            content: ({ payable }) => {
+              if (payable.__typename === 'Invoice') {
```

Comment:
I understand this value is not accessible from another attribute right?
Is there a way to have such returned from BE in a new attribute? Could be a graphQl api only attribute maybe
I always find it dangerous to rely on the `typeName`